tools/db_stress.cc

```diff
@@ -1960,9 +1994,23 @@ class StressTest {
         // will later read the same key before releasing the snapshot and verify
         // that the results are the same.
         auto status_at = db_->Get(ropt, column_family, key, &value_at);
+        std::vector<bool> *key_vec = nullptr;
+
+        if (FLAGS_compare_full_db_state_snapshot &&
+            (thread->tid == 0)) {
```

Comment:
what's the idea behind only tracking state if the snapshot was taken in thread 0?

Reply:
that was meant to reduce memory overhead. Since full snapshot comparison is expensive I chose to do it only in thread 0